src/Raven.Server/Documents/Revisions/RevisionsBinCleanMergedCommand.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.ServerWide.Context;
+using Voron;
+using static Raven.Server.Documents.RevisionsBinCleaner;
+
+namespace Raven.Server.Documents.Revisions;
+
+public partial class RevisionsStorage
+{
+    internal sealed class RevisionsBinCleanMergedCommand : DocumentMergedTransactionCommand
+    {
+        public DateTime _before;
+
+        public long _maxTotalDeletes;
+
+        public long _maxReadsPerBatch;
+
+        public (long DeletedEntries, bool HasMore)? Result { get; private set; }
+
+        public RevisionsBinCleanMergedCommand(DateTime before, long maxTotalDeletes, long maxReadsPerBatch)
+        {
+            _before = before;
+            _maxTotalDeletes = maxTotalDeletes;
+            _maxReadsPerBatch = maxReadsPerBatch;
+        }
+
+        protected override long ExecuteCmd(DocumentsOperationContext context)
+        {
+            var lastEtag = DocumentsStorage.ReadLastRevisionsBinCleanerState(context.Transaction.InnerTransaction);
+            var currentState = new RevisionsBinCleanerState() { LastEtag = lastEtag };
+            Result = DeleteRevisions(context, currentState);
+            if(currentState.LastEtag != lastEtag)
+                context.DocumentDatabase.DocumentsStorage.SetLastRevisionsBinCleanerState(context, currentState.LastEtag);
+            return 1;
+        }
+
+        private (long DeletedRevisions, bool HasMore)? DeleteRevisions(DocumentsOperationContext context, RevisionsBinCleanerState state)
+        {
+            if (_maxReadsPerBatch == 0)
+                return (0, false);
+
+            var revisionsStorage = context.DocumentDatabase.DocumentsStorage.RevisionsStorage;
+            var revisions = revisionsStorage.GetRevisionsBinEntries(context, _before, DocumentFields.Id | DocumentFields.ChangeVector, state); // forget about enumerator
```

Comment:
do we have a test for such case?

Reply:
It will come to the first relevant revision (first `Delete Revision` which is also last revision), and if its `LastModified` is greater then `before`, then the loop will break but its etag will be stored in `RevisionsBinCleanerLastEtag`,
then the cleaner will wait '_configuration.RefreshFrequency' (default value: 5 minutes), and it'll try again..

Reply:
I added new test for this